packages/flutter_test/lib/src/finders.dart

```diff
@@ -548,26 +572,66 @@ abstract class MatchFinder extends Finder {
 }
 
 class _TextFinder extends MatchFinder {
-  _TextFinder(this.text, { bool skipOffstage = true }) : super(skipOffstage: skipOffstage);
+  _TextFinder(
+    this.text, {
+    this.skipRichText = true,
+    bool skipOffstage = true,
+  }) : super(skipOffstage: skipOffstage);
 
   final String text;
 
+  /// Whether standalone [RichText] widgets should be skipped or not.
+  ///
+  /// Defaults to `true`.
+  ///
+  /// If enabled, only [Text] widgets will be considered. These of course also
+  /// insert a [RichText] widget, however, that is ignored and so are all
+  /// standalone [RichText] widgets.
+  /// If disabled, only [RichText] widgets will be considered. This *implicitly*
+  /// considers [Text] widgets as well since they always insert a [RichText]
+  /// child.
+  ///
+  /// In either case, [EditableText] widgets will also be considered.
+  final bool skipRichText;
+
   @override
   String get description => 'text "$text"';
 
   @override
   bool matches(Element candidate) {
     final Widget widget = candidate.widget;
+    if (widget is EditableText)
+      return _matchesEditableText(widget);
+
+    if (skipRichText)
+      return _matchesNonRichText(widget);
+    // It would be sufficient to always use _matchesRichText if we wanted to
+    // match both standalone RichText widgets as well as Text widgets. However,
+    // the find.text() finder used to always ignore standalone RichText widgets,
+    // which is why we need the _matchesNonRichText method in order to not be
+    // backwards-compatible and not break existing tests.
+    return _matchesRichText(widget);
+  }
+
+  bool _matchesRichText(Widget widget) {
+    if (widget is RichText)
+      return widget.text.toPlainText() == text;
+    return false;
+  }
+
+  bool _matchesNonRichText(Widget widget) {
     if (widget is Text) {
       if (widget.data != null)
         return widget.data == text;
       assert(widget.textSpan != null);
       return widget.textSpan!.toPlainText() == text;
```

Comment:
To further explain: in my opinion, this code is completely **redundant** as directly matching `RichText` is the way better approach (as all `Text` widgets insert a `RichText` child).

However, we sadly **have** to keep this old code as we would otherwise break user code...